test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -244,6 +246,32 @@ class CompositeFilterIntegrationTest : public testing::TestWithParam<Network::Ad
     BaseIntegrationTest::createUpstreams();
     addFakeUpstream(Http::CodecType::HTTP2);
   }
+
+  void testBasic(bool downstream_filter) {
+    downstream_filter_ = downstream_filter;
+    prependCompositeFilter();
+    setUpstreamProtocol(Http::CodecType::HTTP2);
+    initialize();
+
+    codec_client_ = makeHttpConnection(lookupPort("http"));
+
+    {
+      auto response = codec_client_->makeRequestWithBody(default_request_headers_, 1024);
+      waitForNextUpstreamRequest();
+
+      upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
+      ASSERT_TRUE(response->waitForEndStream());
+      EXPECT_THAT(response->headers(), Http::HttpStatusIs("200"));
+    }
+
+    {
+      auto response = codec_client_->makeRequestWithBody(match_request_headers_, 1024);
+      ASSERT_TRUE(response->waitForEndStream());
+      EXPECT_THAT(response->headers(), Http::HttpStatusIs("403"));
+    }
+  }
+
+  bool downstream_filter_{true};
```

Comment:
Can you make a test suite parameter, such that all tests in the suite are tested with downstream and upstream configuration?

Reply:
Done!